### PR TITLE
Add component for payButton element

### PR DESCRIPTION
**Pull request by awalker-stripe**
### Summary & motivation

Adds component for the `payButton` element.

[this template]:
  https://github.com/stripe/react-stripe-js/tree/master/.github/API_REVIEW.md

### Testing & documentation

Performed manual testing using a local React project.

### CODE REVIEW

File: src/components/createElementComponent.test.tsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ import {
   PaymentRequestButtonElementComponent,
   CartElementComponent,
 } from '../types';
+import {PayButtonElement} from '..';
```
Comment (jackieosborn-stripe) [inline src/components/createElementComponent.test.tsx:13]: Should we instead do import `PayButtonElementComponent` in the above import line?

File: src/components/createElementComponent.test.tsx [hunk 2]
```diff
@@ -534,6 +551,98 @@ describe('createElementComponent', () => {
...
+        <Elements stripe={mockStripe}>
+          <PayButtonElement onConfirm={mockHandler2} />
+        </Elements>
+      );
+
+      const confirmEventMock = Symbol('lineitemclick');
+      simulateConfirm(confirmEventMock);
+      expect(mockHandler2).toHaveBeenCalledWith(confirmEventMock);
+      expect(mockHandler).not.toHaveBeenCalled();
+    });
```
Comment (jackieosborn-stripe) [inline src/components/createElementComponent.test.tsx:572]: Maybe an accidental copy-paste?

File: src/types/index.ts [hunk 3]
```diff
@@ -431,6 +431,62 @@ export interface PaymentElementProps extends ElementProps {
...
+
+  /**
+   * Triggered when a button on the Element is clicked.
+   */
+  onClick?: (event: stripeJs.StripePayButtonElementClickEvent) => any;
+
+  /**
+   * Triggered when a buyer authorizes a payment within a supported payment method.
+   */
+  onConfirm: (event: stripeJs.StripePayButtonElementConfirmEvent) => any;
```
Comment (jackieosborn-stripe) [inline src/types/index.ts:464]: nice :)

**Review by jackieosborn-stripe — APPROVED** [review]

File: src/components/createElementComponent.test.tsx (hunk 2 above)
Comment (awalker-stripe) [reply src/components/createElementComponent.test.tsx:572]: oops

**Review by jackieosborn-stripe — APPROVED** [review]

**Pull request merged**